MM-25657 Add 'config migrate' (#4206)
mattermost/docs: source/administration/mmctl-cli-tool.rst

   --local                       allows communicating with the server through a unix socket
   --strict                      will only run commands if the mmctl version matches the server one

mmctl config migrate
^^^^^^^^^^^^^^^^^^^^

**Description**

> Migrates a file-based configuration to (or from) a database-based configuration. Point the Mattermost server at the target configuration to start using it.

Reviewer: @isacikgoz Does the server need to be restarted to apply the changes or can it be done without a restart? The config in database docs specify a restart is required https://docs.mattermost.com/administration/config-in-database.html#reload-systemd-files-and-restart-mattermost

Reviewer 2: Actually this command only migrates the configuration data from one to another. To change the store type (file or a DB) the administrator needs to set `MM_CONFIG` environment variable and restart the server app.

Author: @wiersgallak Any additional updates needed for this PR from your perspective?

Reviewer: @cwarnermm I think we should add the additional details about changing the store type as well.  If the admin is migrating, they likely want to have the changes be stored and actually use it ongoing.

Author: Thank you both -- this was a very good clarification to include in the product documentation.

**Format**

.. code-block:: sh

   mmctl config migrate [from_config] [to_config] [flags]

**Examples**

.. code-block:: sh

   config migrate path/to/config.json "postgres://mmuser:mostest@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10"

**Options**

.. code-block:: sh

   -h, --help   help for migrate

**Options inherited from parent commands**

.. code-block:: sh

   --config-path string           path to the configuration directory. If "$HOME/.mmctl" exists it will take precedence over the default value (default "$XDG_CONFIG_HOME")
   --format string                the format of the command output [plain, json] (default "plain")
   --insecure-sha1-intermediate   allows to use insecure TLS protocols, such as SHA-1
   --local                        allows communicating with the server through a unix socket
   --strict                       will only run commands if the mmctl version matches the server one

mmctl config reload
^^^^^^^^^^^^^^^^^^^

**Description**

> Reloads the server configuration in case you want new settings to be applied.

Reviewer 3: ```suggestion
Reloads the server configuration and applies new settings.
```

Reviewer 3: Is my interpretation correct? If so, my suggested wording isn't ideal and is open to refinement. :)
